google/cloud/firestore_v1/async_document.py

```diff
@@ -270,7 +270,7 @@ async def delete(self, option=None):
             still return the time that the request was received by the server.
         """
         write_pb = _helpers.pb_for_delete(self._document_path, option)
```

Comment:
I wasn't on your earlier reviews but could you speak to how much of this code is duplicated from the existing document.py?

Reply:
There's a fair amount of duplication. As far as shared functionality goes, most was refactored to the base class, but instances where rpc calls are made with the same parameters but with an async endpoint, there isn't much we can do to avoid duplication of parameters.

Reply:
If construction of parameters is in any way complex, we can refactor construction to a helper function in the base class.